homeassistant/components/stiebel_eltron/climate.py

```diff
@@ -0,0 +1,153 @@
+import logging
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    ATTR_CURRENT_HUMIDITY, STATE_AUTO, STATE_IDLE, STATE_MANUAL,
+    SUPPORT_OPERATION_MODE, SUPPORT_TARGET_TEMPERATURE)
+from homeassistant.components.stiebel_eltron import DOMAIN as STE_DOMAIN
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+DEPENDENCIES = ['stiebel_eltron']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE
+
+STATE_DAYMODE = 'day_mode'
+STATE_SETBACK = 'setback_mode'
+STATE_DHW = 'dhw'
+STATE_EMERGENCY = 'emergency'
+
+"""Mapping STIEBEL ELTRON states to homeassistant states."""
+STE_TO_HA_STATE = {'AUTOMATIC': STATE_AUTO,
+                   'MANUAL MODE': STATE_MANUAL,
+                   'STANDBY': STATE_IDLE,
+                   'DAY MODE': STATE_DAYMODE,
+                   'SETBACK MODE': STATE_SETBACK,
+                   'DHW': STATE_DHW,
+                   'EMERGENCY OPERATION': STATE_EMERGENCY}
+
+"""Mapping homeassistant states to STIEBEL ELTRON states."""
+HA_TO_STE_STATE = {value: key for key, value in STE_TO_HA_STATE.items()}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the StiebelEltron platform."""
+    name = hass.data[STE_DOMAIN]['name']
+    ste_data = hass.data[STE_DOMAIN]['ste_data']
+
+    add_devices([StiebelEltron(name, ste_data)], True)
+    return True
+
+
+class StiebelEltron(ClimateDevice):
+    """Representation of a STIEBEL ELTRON heat pump."""
+
+    def __init__(self, name, ste_data):
+        """Initialize the unit."""
+        self._name = name
+
+        self._target_temperature = None
+        self._current_temperature = None
+        self._current_humidity = None
+        self._operation_modes = [STATE_AUTO, STATE_IDLE, STATE_DHW]
+        self._current_operation = None
+        self._filter_alarm = None
+        self._ste_data = ste_data
+
+        _LOGGER.debug("Initialized climate platform")
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    def update(self):
+        """Update unit attributes."""
+        self._ste_data.update()
+
+        self._target_temperature = self._ste_data.api.get_target_temp()
+        self._current_temperature = self._ste_data.api.get_current_temp()
+        self._current_humidity = self._ste_data.api.get_current_humidity()
+        self._filter_alarm = self._ste_data.api.get_filter_alarm_status()
+        self._current_operation = self._ste_data.api.get_operation()
+
+        _LOGGER.debug("Update %s, current temp: %s", self._name,
+                      self._current_temperature)
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        return {
+            ATTR_CURRENT_HUMIDITY: self._current_humidity,
```

Comment:
This is part of the base climate entity state attributes already. Implement `current_humidity` property.

Reply:
Done